ror_test/app/controllers/colleges_controller.rb

```diff
@@ -0,0 +1,53 @@
+class CollegesController < ApplicationController
+    def new
+        @college = College.new
+    end
+
+    def index
+        if params[:search]
+            @colleges = College.where('name LIKE ?', "%#{params[:search]}%")
+        else
+            @colleges = College.all
+        end
+     end
+
+    def edit
+        @college = College.find(params[:id])
+    end
+
+    def show
+        @college = College.find(params[:id])
+        @students = Student.select('name', 'id').where('college_name LIKE ?', "%#{@college.name}%")
+    end
+
+    def create
+        @college = College.new(college_params)
+        if @college.save
+            redirect_to @college
+        else
+            render 'new'
+        end
+     end
+
+    def update
+        @college = College.find(params[:id])
```

Comment:
see if this is right approach, what if I pass non existing id value, is the case handled?